firedrake/supermeshimpl.pyx

```diff
@@ -98,3 +103,50 @@ def assemble_mixed_mass_matrix(V_A, V_B, candidates,
 
     CHKERR(MatAssemblyBegin(mat.mat, MAT_FINAL_ASSEMBLY))
     CHKERR(MatAssemblyEnd(mat.mat, MAT_FINAL_ASSEMBLY))
+
+
+def intersection_finder(mesh_A, mesh_B):
+    # Plan:
+    # Call libsupermesh_sort_intersection_finder_set_input
+    # Call libsupermesh_sort_intersection_finder_query_output
+    # Call libsupermesh_sort_intersection_finder_get_output
+    # Return the output
+
+    cdef:
+        numpy.ndarray[long, ndim=2, mode="c"] vertex_map_A, vertex_map_B
+        numpy.ndarray[double, ndim=2, mode="c"] vertices_A, vertices_B
+        long nindices
+        numpy.ndarray[long, ndim=1, mode="c"] indices, indptr
+        long nnodes_A, nnodes_B, ncells_A, ncells_B
+        int dim_A, dim_B, loc_A, loc_B
+
+    vertices_A = mesh_A.coordinates.dat.data_ro_with_halos
+    vertices_B = mesh_B.coordinates.dat.data_ro_with_halos
+    vertex_map_A = mesh_A.coordinates.cell_node_map().values_with_halo.astype(long)
+    vertex_map_B = mesh_B.coordinates.cell_node_map().values_with_halo.astype(long)
+    nnodes_A = mesh_A.num_vertices()
+    nnodes_B = mesh_B.num_vertices()
+    dim_A = mesh_A.geometric_dimension()
+    dim_B = mesh_B.geometric_dimension()
+    ncells_A = mesh_A.num_cells()
+    ncells_B = mesh_B.num_cells()
+    loc_A = vertex_map_A.shape[1]
+    loc_B = vertex_map_B.shape[1]
```

Comment:
What shape of elements does this handle?

Reply:
Does it work if the elements are embedded?

Reply:
Does it work if the mesh is not affine-mapped/bilinearly mapped.

Reply:
It does not work if the mesh is embedded. I have added asserts to check that this is not the case.

It does not work if the mesh is not mapped with degree 1. I have added asserts.

Regarding shape of elements: it works for any polygon/polyhedron (it just relies on bounding box information). The interface does assume that all cells are the same kind of polytope. I have added a test for quads.

Reply:
I noticed that the embedding dimension check is already handled in the calling code (sorry!).

Reply:
I guess it doesn't hurt to check again in case someone calls this directly.

Reply:
Yes, I think other code may want to call `intersection_finder` directly.